sway-lib-std/src/storage/storage_map.sw

```diff
@@ -125,4 +126,67 @@ where
         let key = sha256((key, self.slot));
         clear::<V>(key, 0)
     }
+
+    /// Inserts a key-value pair into the map, if a value does not already exist for the key.
+    ///
+    /// # Arguments
+    ///
+    /// * `key`: [K] - The key to which the value is paired.
+    /// * `value`: [V] - The value to be stored.
+    ///
+    /// # Returns
+    ///
+    /// * [Option<V>] - The value previously stored at `key`, if any.
+    ///
+    /// # Number of Storage Accesses
+    ///
+    /// * Reads: `1`
+    /// * Writes: `1`
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     map: StorageMap<u64, bool> = StorageMap {}
+    /// }
+    ///
+    /// fn foo() {
+    ///     let key = 5_u64;
+    ///     let value = true;
+    ///     storage.map.insert(key, value);
+    ///     let retrieved_value = storage.map.get(key).read();
+    ///     assert(value == retrieved_value);
+    ///
+    ///     let new_value = false;
+    ///     let old_value = storage.map.try_insert(key, new_value);
+    ///     assert(old_value == Option::Some(value)); // The old value is returned.
+    ///     let retrieved_value = storage.map.get(key).read();
+    ///     assert(value == retrieved_value); // New value was not inserted, as a value already existed.
+    ///
+    ///     let key2 = 10_u64;
+    ///     let old_value = storage.map.try_insert(key2, new_value);
+    ///     assert(old_value == Option::None); // No old value is returned.
+    ///     let retrieved_value = storage.map.get(key2).read();
+    ///     assert(new_value == retrieved_value); // New value was inserted, as no value existed prior.
+    /// }
+    /// ```
+    #[storage(read, write)]
+    pub fn try_insert(self, key: K, value: V) -> Option<V>
```

Comment:
I'm unsure about the need to return the previous value.
Why are we returning anything?
I think it's better to return a result to indicate if we have or have not successfully inserted so the user can decide what to do.
An option doesn't make sense to me here

Reply:
Changed it to result and made the return similar to rust